hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1102,7 +1103,17 @@ def as_strategy(strat_or_callable, thing, final=True):
             f"Could not resolve {thing!r} to a strategy, because it is an abstract "
             "type without any subclasses. Consider using register_type_strategy"
         )
-    return sampled_from(subclasses).flatmap(from_type)
+    subclass_strategies = nothing()
```

Comment:
Will `nothing()` be removed from possible strategies if we have other `subclass_strategies`?